### PR TITLE
[logging] fix log race condition

### PR DESCRIPTION
**Pull request by dntczdx**
It guards "logParams". The "logParams" has a chance to be updated while another thread is trying to send it to log server.

### CODE REVIEW

File: cli_tools/common/utils/logging/service/log_entry.go [hunk 1]
```diff
@@ -172,6 +172,9 @@ type OutputInfo struct {
 }
 
 func (i *InputParams) updateParams(p *string) {
+	logParamsMutex.Lock()
```
Comment (zoran15) [inline cli_tools/common/utils/logging/service/log_entry.go]: Why is this called update params when it's updating project info?

Should p called project?

**Comment by EricEdens** [comment]
Can you add some more info about the race condition that you've observed, and which resources you're trying to guard access to?

File: cli_tools/common/utils/logging/service/log_entry.go (hunk 1 above)
Comment (dntczdx) [reply cli_tools/common/utils/logging/service/log_entry.go]: There can be more params updated if necessary. I won't provide multiple functions for that.
Let me adjust the project param name

File: cli_tools/common/utils/logging/service/log_entry.go [hunk 2]
```diff
@@ -171,12 +171,15 @@ type OutputInfo struct {
 	FailureMessageWithoutPrivacyInfo string `json:"failure_message_without_privacy_info,omitempty"`
 }
 
-func (i *InputParams) updateParams(p *string) {
-	if p == nil {
+func (i *InputParams) updateParams(projectPointer *string) {
```
Comment (zoran15) [inline cli_tools/common/utils/logging/service/log_entry.go]: project is enough as it's obvious it's pointer from the type

**Comment by dntczdx** [comment]
> Can you add some more info about the race condition that you've observed, and which resources you're trying to guard access to?

As the mutex's name said, it guards "logParams". The "logParams" has a chance to be updated while another thread is trying to send it to log server.

File: cli_tools/common/utils/logging/service/log_entry.go (hunk 2 above)
Comment (dntczdx) [reply cli_tools/common/utils/logging/service/log_entry.go]: just in order to letting project & obfuscatedProject having the same type...

**Comment by EricEdens** [comment]
> > Can you add some more info about the race condition that you've observed, and which resources you're trying to guard access to?
> 
> As the mutex's name said, it guards "logParams". The "logParams" has a chance to be updated while another thread is trying to send it to log server.

What are `logParams`? An instance of the InputParams struct? If so, why not put your mutex as a field within that struct?

**Pull request merged**